flytekit/core/type_engine.py

```diff
@@ -565,9 +597,17 @@ def get_sub_type(t: Type[T]) -> Type[T]:
         """
         Return the generic Type T of the List
         """
-        if hasattr(t, "__origin__") and t.__origin__ is list:  # type: ignore
-            if hasattr(t, "__args__"):
-                return t.__args__[0]  # type: ignore
+
+        if hasattr(t, "__origin__"):
+
+            # Handle annotation on list generic, eg:
+            # typing.Annotated[typing.List[int, 'foo']]
+            if isinstance(t, typing._AnnotatedAlias) and hasattr(t.__origin__, "__origin__"):
```

Comment:
not entirely sure i understand - why is the checking here using isinstance of the annotated alias and using get origin above?  Could you add a comment please?

Reply:
sloppy (made the change for @eapolinario comment in one spot but not other)